client/tests/AtVersions.test.js

```diff
@@ -0,0 +1,48 @@
+        /* prettier-ignore */
+        await getPage({ role: 'admin', url: '/test-queue' }, async page => {
+            const openTrayIfClosed = async () => {
+                await page.waitForSelector('button ::-p-text(Manage Assistive Technology Versions)');
+                const isTrayClosed = !!(await page.$('::-p-text(Select an Assistive Technology and manage its versions)'));
+                if (isTrayClosed) {
+                    await page.click('button ::-p-text(Manage Assistive Technology Versions)');
+                    await page.waitForSelector('::-p-text(Select an Assistive Technology and manage its versions)');
+                }
+            };
+            await openTrayIfClosed();
+            await page.click('button ::-p-text(Add a New Version)');
+            await page.waitForSelector('.modal-title ::-p-text(Add a New Version for JAWS)');
+            await page.waitForSelector('.modal-body .form-group:nth-child(1) ::-p-text(Version Number)');
+            await page.waitForSelector('.modal-body .form-group:nth-child(2) ::-p-text(Approximate date of availability)');
+            await page.type('.modal-body .form-group:nth-child(1) input', '99.0.1');
+            await page.type('.modal-body .form-group:nth-child(2) input', '01-01-2000');
+            await page.click('.modal-footer button ::-p-text(Add Version)');
+            await page.waitForNetworkIdle();
+            await openTrayIfClosed();
+            await page.waitForSelector('.at-versions-container option:nth-child(2) ::-p-text(99.0.1)');
+            const optionValue = await page.$eval('.at-versions-container option:nth-child(2)', option => option.value);
+            await page.select('.at-versions-container select', optionValue);
+            await page.click('.at-versions-container button ::-p-text(Edit)');
+            const input = await page.waitForSelector('.modal-body .form-group:nth-child(1) input');
+            for (let i = 0; i < 6; i += 1) {
+                await input.press('Backspace');
+            }
+            await page.type('.modal-body .form-group:nth-child(1) input', '99.0.99');
+            await page.click('.modal-footer button ::-p-text(Save)');
+            await page.waitForNetworkIdle();
+            await openTrayIfClosed();
+            await page.waitForSelector('.at-versions-container option ::-p-text(99.0.99)');
+            await page.select('.at-versions-container select', optionValue);
+            await page.click('.at-versions-container button ::-p-text(Remove)');
+            await page.waitForSelector('.modal-title ::-p-text(Remove JAWS Version 99.0.99)');
+            await page.click('.modal-footer button ::-p-text(Remove)');
+            await page.waitForNetworkIdle();
+            await openTrayIfClosed();
+            const option = await page.$('.at-versions-container option ::-p-text(99.0.99)');
+            expect(option).toBeNull();
+        });
+    });
+});
```

Comment:
You know, what I was imagining, if someone needed to come along later and figure out what's happening, is that they would use breakpoints and actually take a look at the real page shown in the browser for each step of the test. This was a motivation for putting extra care into the debugging support for end-to-end tests. I did experiment with creating variables for some of these selectors, but I found that it didn't create a more readable test and in fact may have made it even more intimidating by adding indirection and making it less of a step-by-step "recipe," in addition to making the test quite a bit longer.

With readability in mind, I did try choose selectors not just to point at the elements in question, but also to tell a bit of a story about what is happening. For example, I had `.modal-title ::-p-text(Add a New Version for JAWS)` which hopefully conveys the idea that we're searching inside a modal's title for the text "Add a New Version for JAWS."

Reply:
Understood and I appreciate the care already taken towards authoring these test suites in that regard. I could also be alone in my current thought on it 😅.

But certainly curious about any others' thoughts in the future in case there is a middle ground here!